src/components/Stake.svelte

```diff
@@ -0,0 +1,163 @@
+  storeCurrentAction.subscribe((val) => (showSelf = val == ActionForms.Stake));
+  storeProviderId.subscribe((val) => (providerId = val));
+  storeValidAccounts.subscribe((val) => (validAccounts = val));
+  storeToken.subscribe((val) => (token = val));
+
+  const hideSelf = () => {
+    storeCurrentAction.update((val) => (val = ActionForms.NoForm));
+    clearTxnStatuses();
+  };
+
+  const addNewTxnStatus = (txnStatus: string) => {
+    txnStatuses = [...txnStatuses, txnStatus];
+  };
+  const clearTxnStatuses = () => (txnStatuses = new Array<string>());
+
+  const stake = async (evt: Event) => {
+    evt.preventDefault();
+    clearTxnStatuses();
+    let endpointURI: string = thisDotApi.selectedEndpoint || '';
+    if (selectedKey === '') {
+      alert('Please choose a key to stake from.');
+    } else if (isFunction(thisWeb3FromSource) && isFunction(thisWeb3Enable)) {
+      let signingKeys = validAccounts[selectedKey];
+      showTransactionStatus = true;
+      if (isLocalhost(endpointURI)) {
+        await submitStake(
+          thisDotApi.api as ApiPromise,
+          undefined,
+          signingKeys,
+          providerId,
+          stakeAmountInDollars,
+          endpointURI as string,
+          addNewTxnStatus
+        );
+      } else {
+        const extensions = await thisWeb3Enable('Frequency parachain provider dashboard: Adding Keys');
+        if (extensions.length !== 0) {
+          const injectedExtension = await thisWeb3FromSource(signingKeys.meta.source);
+          await submitStake(
+            thisDotApi.api as ApiPromise,
+            injectedExtension,
+            signingKeys,
+            providerId,
+            stakeAmountInDollars,
+            endpointURI as string,
+            addNewTxnStatus
+          );
+        }
+      }
+    }
+  };
+
+  function handleInput(evt: Event) {
+    const target = evt.target as HTMLInputElement;
+    if (target !== null && target.value === '') {
+      stakeAmount = 0n;
+    } else if (target !== null && target.value !== null) {
+      stakeAmount = BigInt(target.value);
+      return;
+    }
+  }
+</script>
+
+<div class:hidden={!(connected && showSelf)}>
+  <h3>Stake to Provider Id {providerId}</h3>
+  <div class="directions">
+    <p><strong>Directions</strong></p>
+    <ol>
+      <li>Ensure the control key has a FRQCY balance.</li>
+      <li>Click 'Stake'</li>
+      <li>This will require 1 signature to send the transaction.</li>
+    </ol>
+  </div>
+  <form>
+    <KeySelection
+      component="SelectControlKey"
+      selectLabel="Key to Stake From"
+      bind:selectedOption={selectedKey}
+      {validAccounts}
+    />
+    <label for="stakingInput">Amount to Stake in Tokens</label>
+    <div class="input-container">
+      <input type="number" id="stakingInput" value="1" style="text-align: right;" on:input={handleInput} />
+      <span class="units">{token}</span>
+    </div>
+    <button on:click={stake}>Stake</button>
```

Comment:
```suggestion
    <button on:click|preventDefault={stake}>Stake</button>
```
This is the (Svelte) Way

Reply:
Ah, right. I knew that, but chatGPT led me astray.